apps/site/components/JSX/CodeBox/index.tsx

```diff
@@ -0,0 +1,27 @@
+import dedent from 'dedent';
+import { toJsxRuntime } from 'hast-util-to-jsx-runtime';
+import type { FC, PropsWithChildren } from 'react';
+
+import MDXCodeBox from '@/components/MDX/CodeBox';
+import { reactRuntime } from '@/next.mdx.evaluater.mjs';
+import { highlightToHast, shikiPromise } from '@/util/getHighlighter';
+
+type CodeBoxProps = { language: string; showCopyButton?: string };
+
+const codeToHast = highlightToHast(await shikiPromise);
+
+const CodeBox: FC<PropsWithChildren<CodeBoxProps>> = async props => {
+  const out = codeToHast(dedent(props.children as string), props.language);
+  const className = `language-${props.language}`;
+
+  return toJsxRuntime(out, {
```

Comment:
I realize this bypassed our MDX Rehype Plugin, unironically. We support three types of CodeBoxes now:

- MDXCodeBox, source comes from MDX code blocks and code snippet will be transpilled within the MDX parsing process.
- CodeBox, plain base CodeBox component without syntax highlighting
- JSXCodeBox, the source is a children string/ReactNode, and the code snippet gets transpilled directly within the component.

I assume the difference between the MDX and JSX is that the MDX requires plain code blocks within a Markdown/MDX file, and the JSX one uses a JSX component within MDX but does the transpilation outside of MDX.

Both use the same memoized Shiki instance.